### DISCUSSION TITLE
Server Requirements for Docker

### DISCUSSION THREAD

**Discussion by SgiobairOg** [coqui-ai/TTS#530] [Ideas]
Hello, I've been looking at getting the server up and running in a docker image for some experiments at work. 

I have things working no problem but I've noticed the image I'm generating is very large with one layer along taking 4.68 GB. 

Wondering if you can help me identify what needs to be copied over for the server to run with multi-model support and what can be left behind. I know at the moment I'm copying over everything and I expect the notebooks, tests, and some other stuff can be left behind. I'm also wondering if I can trim down the dependencies a bit if we won't be making any models. 

I'm a bit new to python servers (my server experience is mostly node) but if there's some changes needed there to support this and if you're interested in having an image as part of the project again I'd be happy to contribute the development.

Depending on how much time I can invest I'm also considering adding an API so we can separate out our UIs.

Thanks for sharing your project!

**Comment by SgiobairOg** [comment]
And apologies, if I'm looking at it properly it does appear that there is already an API.

**Reply by thorstenMueller** [reply]
Hello @SgiobairOg .
So your TTS server is running, web ui is reachable on port 5002 and you can synthesize voice with web ui?
Then you can call the api this way:

`curl <server>:5002/api/tts?text=Hallo --output=out.wav`

**Reply by nauen** [reply]
curl -o out.wav http://localhost:5002/api/tts?text=Hallo
